From 8cb085b755d26f4f86ff48cc7c13acd5b31c47b0 Mon Sep 17 00:00:00 2001
From: Benoit Rat <benoit@sevensols.com>
Date: Fri, 26 Jul 2013 14:18:13 +0200
Subject: [PATCH 1/6] Add patch to fix "gets" firing error during m4
 compilation

This is a patch of patch applied to m4 library during buildroot compilation.
This error occurs with recent GCC version (4.7.3 on ubuntu 13.04).
---
 package/m4/m4-1.4.15-fix_gets_firing_error.patch | 27 ++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 package/m4/m4-1.4.15-fix_gets_firing_error.patch

diff --git a/package/m4/m4-1.4.15-fix_gets_firing_error.patch b/package/m4/m4-1.4.15-fix_gets_firing_error.patch
new file mode 100644
index 0000000..d5a649b
--- /dev/null
+++ b/package/m4/m4-1.4.15-fix_gets_firing_error.patch
@@ -0,0 +1,27 @@
+From e6f044da4b928e6ac48743732e1764a1fe9f8a00 Mon Sep 17 00:00:00 2001
+From: Benoit Rat <benoit@sevensols.com>
+Date: Fri, 26 Jul 2013 11:38:02 +0200
+Subject: [PATCH] gets generate error on new compiler
+
+---
+ lib/stdio.in.h | 4 +++-
+ 1 file changed, 3 insertions(+), 1 deletion(-)
+
+diff --git a/lib/stdio.in.h b/lib/stdio.in.h
+index e98378a..98d03aa 100644
+--- a/lib/stdio.in.h
++++ b/lib/stdio.in.h
+@@ -141,7 +141,9 @@ _GL_WARN_ON_USE (fflush, "fflush is not always POSIX compliant - "
+    so any use of gets warrants an unconditional warning.  Assume it is
+    always declared, since it is required by C89.  */
+ #undef gets
+-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
++#if HAVE_RAW_DECL_GETS 
++   _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
++#endif 
+ 
+ #if 1
+ # if 0
+-- 
+1.8.1.2
+
-- 
2.1.4

